server.go

```diff
@@ -352,6 +364,26 @@ func WithoutLogger() func(*Server) {
 	}
 }
 
+// WithListener configures the server to use a custom listener.
+func WithListener(listener net.Listener) func(*Server) {
+	return func(s *Server) {
+		if s.listener != nil {
+			panic("a listener is already configured; cannot overwrite it")
+		}
+		s.isTLS = isTLSListener(listener)
+		WithAddr(listener.Addr().String())(s)
```

Comment:
The one check that may be worth adding. Is the check if the listener `Addr` and the `Server.Server.Addr` match. If they don't `slog.Warn("WithAddr and provided listener addr do match. Default to listen address")` or something along those lines.